auslib/web/admin/swagger/api.yaml

```diff
@@ -232,9 +957,182 @@ definitions:
+        $ref: "#/definitions/UserPermissionsModel"
+
+      release_locale:
+        description: "Release Locale permission to perform modify action only on product releases."
+        allOf:
+          - $ref: "#/definitions/DataVersionModel"
+          - type: object
+            properties:
+              options:
+                type: ['object', 'null']
+                description: Options available for the given permission
+                properties:
+                  products:
+                    type: array
+                    uniqueItems: true
+                    items:
+                      type: string
+                  actions:
+                    type: array
+                    uniqueItems: true
+                    items:
+                      type: string
+                      enum: ["modify"]
+
+      release_read_only:
+        description: "Read only Releases permission to perform modify action only on product releases."
+        allOf:
+          - $ref: "#/definitions/DataVersionModel"
+          - type: object
+            properties:
+              options:
+                type: ['object', 'null']
+                description: Options available for the given permission
+                properties:
+                  products:
+                    type: array
+                    uniqueItems: true
+                    items:
+                      type: string
+                  actions:
+                    type: array
+                    uniqueItems: true
+                    items:
+                      type: string
+                      enum: ["set", "unset"]
+
+      permission:
+        description: "permission to perform modify actions only."
+        allOf:
+          - $ref: "#/definitions/DataVersionModel"
+          - type: object
+            properties:
+              options:
+                type: ['object', 'null']
+                description: Options available for the given permission
+                properties:
+                  actions:
+                    type: array
+                    uniqueItems: true
+                    items:
+                      type: string
+                      enum: ["create", "modify", "delete"]
+
+      scheduled_change:
+        description: "permission to schedule a change to rule, release or permission. Only the Balrog Agent should be granted this permission."
+        allOf:
+          - $ref: "#/definitions/DataVersionModel"
+          - type: object
+            properties:
+              options:
+                type: ['object', 'null']
+                description: Options available for the given permission
+                properties:
+                  actions:
+                    type: array
+                    uniqueItems: true
+                    items:
+                      type: string
+                      enum: ["enact"]
+
```

Comment:
It must've been a pain to fully implement all of these permissions schemas, very good work though!